addons/checkparam/checkparam.lua

```diff
@@ -0,0 +1,305 @@
+
+require('logger')
+res = require('resources')
+extdata = require('extdata')
+config = require('config')
+packets = require('packets')
+
+defaults = {
+  WAR = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  MNK = 'store tp|double attack|triple attack|quadruple attack|martial arts|subtle blow',
+  WHM = 'cure potency|cure potency ii|fast cast|cure spellcasting time|enmity|healing magic casting time|divine benison|damage taken|physical damage taken|magic damage taken',
+  BLM = 'magic attack bonus|magic burst damage|magic burst damage ii|int|magic accuracy|magic damage|fast cast|elemental magic casting time',
+  RDM = 'magic attack bonus|magic burst damage|magic burst damage ii|magic accuracy|fast cast|enfeebling magic skill|enhancing magic skill|store tp|dual wield',
+  THF = 'store tp|double attack|triple attack|quadruple attack|dual wield',
+  PLD = 'enmity|damage taken|physical damage taken|magic damage taken|spell inturruption rate|pharanx|cure potency|fastcast',
+  DRK = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  BST = 'pet: double attack|pet: magic attack bonus|pet: damage taken',
+  BRD = 'all songs|song effect duration|fast cast|song spellcasting time',
+  RNG = 'store tp|snapshot|rapid shot|weapon skill damage',
+  SAM = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  NIN = 'store tp|double attack|triple attack|quadruple attack|subtle blow',
+  DRG = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  SMN = 'blood pact delay|blood pact delay ii|blood pact damage|avatar perpetuation cost|pet: magic attack bonus|pet: str|pet: double attack',
+  BLU = 'store tp|double attack|triple attack|quadruple attack|critical hit rate|critical hit damage|weapon skill damage|fast cast|magic attack bonus|magic accuracy|cure potency',
+  COR = 'store tp|snapshot|rapid shot|fast cast|cure potency|magic accuracy|magic attack bonus|magic damage|weapon skill damage',
+  PUP = 'pet: hp|pet: damage taken|pet: regen|martial arts|store tp|double attack|triple attack|quadruple attack',
+  DNC = 'store tp|double attack|triple attack|quadruple attack',
+  SCH = 'magic attack bonus|magic burst damage|magic burst damage ii|magic accuracy|magic damage|fast cast|elemental magic casting time|cure potency|enh mag eff dur|enhancing magic effect duration',
+  GEO = 'pet: regen|pet: damage taken|indicolure effect duration|indi eff dur|fast cast|magic evasion',
+  RUN = 'enmity|damage taken|physical damage taken|magic damage taken|spell inturruption rate|pharanx|inquartata|fastcast',
+  levelfilter = 99,
+  debugmode = false,
+}
+settings = config.load(defaults)
+
+tbl = {}
+
+windower.register_event('addon command',function(arg)
+  if arg == 'debugmode' then
+    settings.debugmode = not settings.debugmode
+    settings:save('all')
+  else
+    for i=0,#res.slots do
+      local slot = windower.regex.replace(string.lower(res.slots[i].en),' ','_')
```

Comment:
Throughout the file you use `en` and `ens`, try to use `english` and `english_short`.